src/Nethermind/Nethermind.Trie/RustVerkleLib.cs

```diff
@@ -16,136 +16,219 @@
 
-        [DllImport("rust_verkle")]
-        private static extern IntPtr verkle_trie_get(IntPtr verkleTrie, byte[] key);
-        
-        [DllImport("rust_verkle")]
-        private static extern IntPtr get_root_hash(IntPtr verkleTrie);
-        
-        [DllImport("rust_verkle")]
-        private static extern void verkle_trie_insert(IntPtr verkleTrie, byte[] key, byte[] value);
-        
-        [DllImport("rust_verkle")]
-        private static extern IntPtr get_verkle_proof(IntPtr verkleTrie, byte[] key);
+    [DllImport("rust_verkle")]
+    private static extern unsafe IntPtr verkle_trie_get(IntPtr verkleTrie, byte *  key);
+    
+    [DllImport("rust_verkle")]
+    private static extern IntPtr get_root_hash(IntPtr verkleTrie);
+    
+    [DllImport("rust_verkle")]
+    private static extern unsafe void verkle_trie_insert(IntPtr verkleTrie, byte * key, byte * value);
+    
+    [DllImport("rust_verkle")]
+    private static extern unsafe IntPtr get_verkle_proof(IntPtr verkleTrie, byte * key);
 
-        [DllImport("rust_verkle")]
-        private static extern byte verify_verkle_proof(IntPtr verkleTrie, byte[] verkleProof, int proof_len, byte[] key, byte[] value);
+    [DllImport("rust_verkle")]
+    private static extern unsafe byte verify_verkle_proof(IntPtr verkleTrie, byte * verkleProof, int proofLen, byte * key, byte * value);
 
-        [DllImport("rust_verkle")]
-        private static extern void verkle_trie_insert_multiple(IntPtr verkleTrie, byte[,] keys, byte[,] vals, int len);
+    [DllImport("rust_verkle")]
+    private static extern unsafe void verkle_trie_insert_multiple(IntPtr verkleTrie, byte * keys, byte * values, int len);
 
-        [DllImport("rust_verkle")]
-        private static extern IntPtr get_verkle_proof_multiple(IntPtr verkleTrie, byte[,] keys, int len);
+    [DllImport("rust_verkle")]
+    private static extern unsafe IntPtr get_verkle_proof_multiple(IntPtr verkleTrie, byte * keys, int len);
 
-        [DllImport("rust_verkle")]
-        private static extern byte verify_verkle_proof_multiple(IntPtr verkleTrie, byte[] verkleProof, int proof_len, byte[,] key, byte[,] value, int len);
+    [DllImport("rust_verkle")]
+    private static extern unsafe byte verify_verkle_proof_multiple(IntPtr verkleTrie, byte * verkleProof, int proofLen, byte * keys, byte * values, int len);
 
-        public static IntPtr VerkleTrieNew()
+    public static IntPtr VerkleTrieNew()
+    {
+        return verkle_trie_new();
+    }
+    
+    public static unsafe void VerkleTrieInsert(IntPtr verkleTrie, Span<byte> key, Span<byte> value)
+    {
+        int valueLength = value.Length;
+        if (valueLength != 32)
         {
-            return verkle_trie_new();
+            throw new InvalidOperationException("Value length must be less than 32");
         }
-        
-        public static void VerkleTrieInsert(IntPtr verkleTrie, byte[] key, byte[] value)
+
+        fixed (byte* pKey = &MemoryMarshal.GetReference(key))
```

Comment:
If key and value are Span from stackalloc then we don't need fixing it, they a re on stack already fixed. If they are on heap (slice of an array) then we need fixing them. Maybe we should check if there is performance penalty here and differentiate between those scenarios?

Reply:
yes, this makes sense. I will check for the performance penalties and update this

Reply:
I am not sure how to check if the Span is on stack?
also the function signature have the data type as Span<byte>, so in the function I am not able to convert he Span<byte> to byte * without using fixed block